ostree-ext/src/tar/write.rs

```diff
@@ -366,22 +366,32 @@ async fn filter_tar_async(
         let dest = tokio_util::io::SyncIoBridge::new(tx_buf);
 
         let r = filter_tar(&mut src, dest, &config, &repo_tmpdir);
+
+        // We need to make sure to flush out the decompressor here,
```

Comment:
I'd say "flush out decompressor and tar stream" because as I said in an earlier comment I believe `filter_tar` could also leave trailing data.